src/grizzly/common/cache.py

```diff
@@ -0,0 +1,109 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from __future__ import annotations
+
+from contextlib import suppress
+from logging import getLogger
+from shutil import move, rmtree
+from time import time
+from typing import TYPE_CHECKING
+
+from .utils import grz_tmp, interprocess_lock
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+
+_ACTIVE_CACHE = None
+CACHE_PATH = grz_tmp("cache")
+CACHE_TIME = int(time())
+LOCK_ID = "cache"
+LOG = getLogger(__name__)
+MAX_AGE = 86400
+
+
+def _active_cache(max_age: int = MAX_AGE) -> Path:
+    """Retrieve the active cache directory. Create one if needed.
+
+    Args:
+        max_age: Maximum age of active cache (relative to process launch time).
+
+    Returns:
+        Directory to use to store and retrieve cached content.
+    """
+    global _ACTIVE_CACHE  # pylint: disable=global-statement
+    assert max_age >= 0
+    if _ACTIVE_CACHE is None:
+        with interprocess_lock(LOCK_ID):
+            limit = CACHE_TIME - max_age
+            # find most recent active entry
+            for entry in sorted(CACHE_PATH.iterdir(), reverse=True):
```

Comment:
This sort is alphabetical, not numeric. Not that we're likely to hit this case (until 2286), but it is wrong.

Reply:
Yep you are right and as we discussed we should also replace this with checking the "creation time" (but that is not currently available on all platforms).

Reply:
Addressed in #482